### ISSUE TITLE
Add hint parameter to 'find_library' call for usrsctp

### ISSUE THREAD

**Issue by eldog** [rawrtc/rawrtc#72]
I'm having trouble compiling using GCC 5 on Arch Linux 64bit. I understand GCC 7 compatibility isn't supported at the moment (from  https://github.com/rawrtc/rawrtc/issues/68).

I have the error when running `make`

```
 error: ‘SCTP_EVENT_READ’ undeclared (first use in this function)
```

Not sure where this is supposed to be declared/included. Or if I've not installed a dependency, to get `cmake` to run I did an edit of the `CMakeList.txt` file as described below, so it may be all my own doing.

**Details**

I made the dependencies
```
CC=gcc-5 ./make-dependencies.sh 
```

Before running `cmake`, I ran
```
export PKG_CONFIG_PATH=${PWD}/build/prefix/lib/pkgconfig
```

However, it couldn't find `libre`, so to get `cmake` to work I had to edit the `CMakeLists.txt` and add 
```
set(ENV{PKG_CONFIG_PATH} "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig/:$ENV{PKG_CONFIG_PATH}")
```

My cmake command was
```
CC=gcc-5 CXX=g++-5 cmake -DCMAKE_INSTALL_PREFIX=${PWD}/prefix ..  
```

System
```
$ uname -a                                                                                                                                           :(
Linux ng 4.12.4-1-ARCH #1 SMP PREEMPT Fri Jul 28 18:54:18 UTC 2017 x86_64 GNU/Linux
```

Log below

```
/usr/bin/cmake -H/home/eldog/Dropbox/experiments/rawrtc -B/home/eldog/Dropbox/experiments/rawrtc/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /home/eldog/Dropbox/experiments/rawrtc/build/CMakeFiles /home/eldog/Dropbox/experiments/rawrtc/build/CMakeFiles/progress.marks
make -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/home/eldog/Dropbox/experiments/rawrtc/build'
make -f src/librawrtc/CMakeFiles/rawrtc-static.dir/build.make src/librawrtc/CMakeFiles/rawrtc-static.dir/depend
make[2]: Entering directory '/home/eldog/Dropbox/experiments/rawrtc/build'
cd /home/eldog/Dropbox/experiments/rawrtc/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/eldog/Dropbox/experiments/rawrtc /home/eldog/Dropbox/experiments/rawrtc/src/librawrtc /home/eldog/Dropbox/experiments/rawrtc/build /home/eldog/Dropbox/experiments/rawrtc/build/src/librawrtc /home/eldog/Dropbox/experiments/rawrtc/build/src/librawrtc/CMakeFiles/rawrtc-static.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/eldog/Dropbox/experiments/rawrtc/build'
make -f src/librawrtc/CMakeFiles/rawrtc-static.dir/build.make src/librawrtc/CMakeFiles/rawrtc-static.dir/build
make[2]: Entering directory '/home/eldog/Dropbox/experiments/rawrtc/build'
[  1%] Building C object src/librawrtc/CMakeFiles/rawrtc-static.dir/sctp_transport.c.o
cd /home/eldog/Dropbox/experiments/rawrtc/build/src/librawrtc && /usr/bin/gcc-5  -I/home/eldog/Dropbox/experiments/rawrtc/build/prefix/include/re -I/home/eldog/Dropbox/experiments/rawrtc/build/prefix/include/rew -I/home/eldog/Dropbox/experiments/rawrtc/src/include  -g   -o CMakeFiles/rawrtc-static.dir/sctp_transport.c.o   -c /home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c
In file included from /home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:17:0:
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c: In function ‘read_event_handler’:
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.h:7:32: error: ‘SCTP_EVENT_READ’ undeclared (first use in this function)
 #define RAWRTC_SCTP_EVENT_ALL (SCTP_EVENT_READ | SCTP_EVENT_WRITE | SCTP_EVENT_ERROR)
                                ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:1632:16: note: in expansion of macro ‘RAWRTC_SCTP_EVENT_ALL’
         return RAWRTC_SCTP_EVENT_ALL;
                ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.h:7:32: note: each undeclared identifier is reported only once for each function it appears in
 #define RAWRTC_SCTP_EVENT_ALL (SCTP_EVENT_READ | SCTP_EVENT_WRITE | SCTP_EVENT_ERROR)
                                ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:1632:16: note: in expansion of macro ‘RAWRTC_SCTP_EVENT_ALL’
         return RAWRTC_SCTP_EVENT_ALL;
                ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.h:7:50: error: ‘SCTP_EVENT_WRITE’ undeclared (first use in this function)
 #define RAWRTC_SCTP_EVENT_ALL (SCTP_EVENT_READ | SCTP_EVENT_WRITE | SCTP_EVENT_ERROR)
                                                  ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:1632:16: note: in expansion of macro ‘RAWRTC_SCTP_EVENT_ALL’
         return RAWRTC_SCTP_EVENT_ALL;
                ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.h:7:69: error: ‘SCTP_EVENT_ERROR’ undeclared (first use in this function)
 #define RAWRTC_SCTP_EVENT_ALL (SCTP_EVENT_READ | SCTP_EVENT_WRITE | SCTP_EVENT_ERROR)
                                                                     ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:1632:16: note: in expansion of macro ‘RAWRTC_SCTP_EVENT_ALL’
         return RAWRTC_SCTP_EVENT_ALL;
                ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c: In function ‘write_event_handler’:
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.h:7:32: error: ‘SCTP_EVENT_READ’ undeclared (first use in this function)
 #define RAWRTC_SCTP_EVENT_ALL (SCTP_EVENT_READ | SCTP_EVENT_WRITE | SCTP_EVENT_ERROR)
                                ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:1707:16: note: in expansion of macro ‘RAWRTC_SCTP_EVENT_ALL’
         return RAWRTC_SCTP_EVENT_ALL;
                ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.h:7:50: error: ‘SCTP_EVENT_WRITE’ undeclared (first use in this function)
 #define RAWRTC_SCTP_EVENT_ALL (SCTP_EVENT_READ | SCTP_EVENT_WRITE | SCTP_EVENT_ERROR)
                                                  ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:1707:16: note: in expansion of macro ‘RAWRTC_SCTP_EVENT_ALL’
         return RAWRTC_SCTP_EVENT_ALL;
                ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.h:7:69: error: ‘SCTP_EVENT_ERROR’ undeclared (first use in this function)
 #define RAWRTC_SCTP_EVENT_ALL (SCTP_EVENT_READ | SCTP_EVENT_WRITE | SCTP_EVENT_ERROR)
                                                                     ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:1707:16: note: in expansion of macro ‘RAWRTC_SCTP_EVENT_ALL’
         return RAWRTC_SCTP_EVENT_ALL;
                ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c: In function ‘error_event_handler’:
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.h:7:32: error: ‘SCTP_EVENT_READ’ undeclared (first use in this function)
 #define RAWRTC_SCTP_EVENT_ALL (SCTP_EVENT_READ | SCTP_EVENT_WRITE | SCTP_EVENT_ERROR)
                                ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:1747:16: note: in expansion of macro ‘RAWRTC_SCTP_EVENT_ALL’
         return RAWRTC_SCTP_EVENT_ALL;
                ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.h:7:50: error: ‘SCTP_EVENT_WRITE’ undeclared (first use in this function)
 #define RAWRTC_SCTP_EVENT_ALL (SCTP_EVENT_READ | SCTP_EVENT_WRITE | SCTP_EVENT_ERROR)
                                                  ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:1747:16: note: in expansion of macro ‘RAWRTC_SCTP_EVENT_ALL’
         return RAWRTC_SCTP_EVENT_ALL;
                ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.h:7:69: error: ‘SCTP_EVENT_ERROR’ undeclared (first use in this function)
 #define RAWRTC_SCTP_EVENT_ALL (SCTP_EVENT_READ | SCTP_EVENT_WRITE | SCTP_EVENT_ERROR)
                                                                     ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:1747:16: note: in expansion of macro ‘RAWRTC_SCTP_EVENT_ALL’
         return RAWRTC_SCTP_EVENT_ALL;
                ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c: In function ‘upcall_handler_helper’:
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:1766:18: warning: implicit declaration of function ‘usrsctp_get_events’ [-Wimplicit-function-declaration]
     int events = usrsctp_get_events(socket);
                  ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:1783:22: error: ‘SCTP_EVENT_ERROR’ undeclared (first use in this function)
         if (events & SCTP_EVENT_ERROR) {
                      ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:1788:22: error: ‘SCTP_EVENT_READ’ undeclared (first use in this function)
         if (events & SCTP_EVENT_READ) {
                      ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:1793:22: error: ‘SCTP_EVENT_WRITE’ undeclared (first use in this function)
         if (events & SCTP_EVENT_WRITE) {
                      ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c: In function ‘timer_handler’:
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:1818:5: warning: implicit declaration of function ‘usrsctp_handle_timers’ [-Wimplicit-function-declaration]
     usrsctp_handle_timers(RAWRTC_SCTP_TRANSPORT_TIMER_TIMEOUT);
     ^
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c: In function ‘rawrtc_sctp_transport_create’:
/home/eldog/Dropbox/experiments/rawrtc/src/librawrtc/sctp_transport.c:2113:9: warning: implicit declaration of function ‘usrsctp_set_upcall’ [-Wimplicit-function-declaration]
     if (usrsctp_set_upcall(transport->socket, upcall_handler_helper, transport)) {
         ^
make[2]: *** [src/librawrtc/CMakeFiles/rawrtc-static.dir/build.make:474: src/librawrtc/CMakeFiles/rawrtc-static.dir/sctp_transport.c.o] Error 1
make[2]: Leaving directory '/home/eldog/Dropbox/experiments/rawrtc/build'
make[1]: *** [CMakeFiles/Makefile2:165: src/librawrtc/CMakeFiles/rawrtc-static.dir/all] Error 2
make[1]: Leaving directory '/home/eldog/Dropbox/experiments/rawrtc/build'
make: *** [Makefile:133: all] Error 2
```

**Comment by eldog** [comment]
I have a system installation of `libusrsctp` installed and it looks like `SCTP_EVENT_READ` isn't in master. I see that `./make_dependencies.sh` uses a different branch of `usrsctp`. I think this may be the source of the problem. It might be not be using the local version.

**Comment by eldog** [comment]
Got it working!

Needed to specify the `CMAKE_INCLUDE_PATH` to ensure it found the prefix version

```
CC=gcc-5 CXX=g++-5 cmake -DCMAKE_INSTALL_PREFIX=${PWD}/prefix -DCMAKE_PREFIX_PATH=${PWD}/prefix ..
```

**Comment by lgrahl** [comment]
I'll reopen this as I consider it a bug. We should [add a *hint* parameter](https://cmake.org/cmake/help/latest/command/find_library.html) to the `find_library` call of usrsctp [here](https://github.com/rawrtc/rawrtc/blob/7c59c8a92fc53f51c08c03127e5ffba1c2354200/CMakeLists.txt#L70), so the prefixed library is being picked up before the system library. It's not needed for the other dependencies as they all use pkg-config. @eldog would you be able to provide a PR to this?

**Comment by lgrahl** [comment]
Resolved by #73
